### PR TITLE
Fix SP check for x64/x86, remove for arm32/arm64

### PR DESCRIPTION
**Pull request by BruceForstall**
The actual checking had gotten lost between JIT32 and RyuJIT.
I only fixed the "on return from function" case. I removed the check
around calls that existed in JIT32 because in RyuJIT it's not
easy to know where the stack arguments begin without introducing
new IR nodes.

Removed arm64 case because it's not easy to store SP to a stack local
or directly compare SP against a stack local without a temporary.
Also, for the fixed outgoing arg space ABIs (all but x86), these
checks don't seem too useful anyway, so I also removed the arm case.

Fixed several p/invoke tests that were incorrectly using stdcall instead
of cdecl.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT x64 Checked Build and Test
@dotnet-bot test Windows_NT x86 Checked Build and Test

File: src/jit/codegencommon.cpp [hunk 1]
```diff
@@ -6304,8 +6304,8 @@ void CodeGen::genZeroInitFrame(int untrLclHi, int untrLclLo, regNumber initReg,
         regMask = genFindLowestBit(availMask);
         rZero2  = genRegNumFromMask(regMask);
         availMask &= ~regMask;
-        assert((genRegMask(rZero2) & intRegState.rsCalleeRegArgMaskLiveIn) ==
-               0); // rZero2 is not a live incoming argument reg
+        assert((genRegMask(rZero2) & intRegState.rsCalleeRegArgMaskLiveIn) == 0); // rZero2 is not a live incoming
```
Comment (mikedn) [inline src/jit/codegencommon.cpp]: Might be good to move these comments before the assert. Having them at the end of the line is kind of weird, especially when you have line length limits and they're enforced with formatting tools.

File: src/jit/codegencommon.cpp [hunk 2]
```diff
@@ -11927,4 +11927,40 @@ void CodeGen::genReturn(GenTree* treeNode)
...
+{
+    if (doStackPointerCheck)
+    {
+        noway_assert(lvaStackPointerVar != 0xCCCCCCCC && compiler->lvaTable[lvaStackPointerVar].lvDoNotEnregister &&
+                     compiler->lvaTable[lvaStackPointerVar].lvOnFrame);
+        getEmitter()->emitIns_S_R(INS_cmp, EA_PTRSIZE, REG_SPBASE, lvaStackPointerVar, 0);
+
+        BasicBlock*  sp_check = genCreateTempLabel();
+        emitJumpKind jmpEqual = genJumpKindForOper(GT_EQ, CK_SIGNED);
+        inst_JMP(jmpEqual, sp_check);
```
Comment (mikedn) [inline src/jit/codegencommon.cpp]: Just use `INS_je` so I don't have to fix this in my JCC PR.

File: src/jit/compiler.hpp [hunk 3]
```diff
@@ -150,9 +150,9 @@ unsigned __int64 genFindHighestBit(unsigned __int64 mask)
 #endif // 0
 
 /*****************************************************************************
-*
-*  Return true if the given 64-bit value has exactly zero or one bits set.
-*/
+ *
```
Comment (mikedn) [inline src/jit/compiler.hpp]: What's up with all these changes?

File: src/jit/codegencommon.cpp (hunk 1 above)
Comment (BruceForstall) [reply src/jit/codegencommon.cpp]: Yeah, it's weird that jit-format decided to reformat all kinds of lines that I didn't touch.

File: src/jit/compiler.hpp (hunk 3 above)
Comment (BruceForstall) [reply src/jit/compiler.hpp]: jit-format doing its magic -- not sure why.

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT x64 Checked Build and Test
@dotnet-bot test Windows_NT x86 Checked Build and Test

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT x64 Checked Build and Test
@dotnet-bot test Windows_NT x86 Checked Build and Test

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT x64 Checked Build and Test
@dotnet-bot test Windows_NT x86 Checked Build and Test

**Comment by BruceForstall** [comment]
cc @dotnet/jit-contrib

File: src/jit/compiler.hpp (hunk 3 above)
Comment (briansull) [reply src/jit/compiler.hpp]: jit-format is also doing thsi for me too

**Pull request merged**